### ISSUE TITLE
Quarto processes several files when previewing one

### ISSUE THREAD

**Issue by allefeld** [quarto-dev/quarto-cli#7049]

**Comment by mcanouil** [comment]
Please fill in properly the issue.
This is not helpful at all like this.

**Comment by cscheid** [comment]
I think I know what the report is about. If, in a project website, some of the other files have changed, then quarto needs to render the changed files because they might have changed titles, which causes a necessary change on the sidebar for _all pages_. 

There's nothing much we can do in this case. If none of the files have changed and quarto is still rendering everything, then this is a bug we need to investigate. In that case, like @mcanouil said, we need more information.

**Comment by allefeld** [comment]
I accidentally pressed Enter in one of the form fields, which posted the incomplete issue. Sorry about that, but this GitHub issue forms functionality seems... inconvenient, if not broken.

While completing the issue, I tried to reproduce it. So far, I haven't been able to consistently do so.

I understand the dependency via titles, but my titles haven't changed. And even if they had, executing code should not be necessary to get the new ones?

**Comment by mcanouil** [comment]
> Sorry about that, but this GitHub issue forms functionality seems... inconvenient, if not broken.

Could you elaborate?

**Comment by allefeld** [comment]
> > Sorry about that, but this GitHub issue forms functionality seems... inconvenient, if not broken.
> 
> Could you elaborate?

As I said, pressing Enter in one of the form fields directly posts the issue. That is unexpected behavior, especially since some of the fields are multiline.

> Do you use freeze and/or caching?

No, at least not intentionally.

> What's the context? Default, book, manuscript, website projects?
> Which IDE?

See now filled in issue.

> (Note that the issue form goal is to reduce the amount of time required to ask those questions)

I find them awkward, but that's not really the point. An issue shouldn't be posted without the user explicitly saying so, not by pressing some button.

**Comment by allefeld** [comment]
@cscheid is there any way to suppress this behaviour? I could live with temporarily outdated titles in the sidebar.

**Comment by cscheid** [comment]
> As I said, pressing Enter in one of the form fields directly posts the issue. That is unexpected behavior, especially since some of the fields are multiline.

I believe this to be a github bug, unfortunately. We have no control over the keyboard behavior that github produces from those.

**Comment by cscheid** [comment]
> I understand the dependency via titles, but my titles haven't changed. And even if they had, executing code should not be necessary to get the new ones?

Yes, it is currently necessary, because titles can include the results executable code by a number of paths, including shortcodes and inline expressions. Yes, in a perfect world we'd have a static analyzer that can skip the recomputation when it's safe to do so, but that's not going to be practical in the foreseeable future.

**Comment by allefeld** [comment]
Thanks!

Yes, it is a GitHub bug, I only elaborated because mcanouil asked me to. Sorry for #7050, please feel free to delete.

**Comment by cderv** [comment]
@allefeld probaby this one but it did not get traction yet
* https://github.com/orgs/community/discussions/10354

**Comment by royfrancis** [comment]
I think this question started off with exactly the question I had and then digressed to titles and github problems. It is possible to **render** one file alone in a **project**. But, when trying to **preview** one file, all the other qmd files in the project seems to be executed/run too. 
So, if I have one completed qmd and several other incomplete qmds, I cannot preview that one complete file alone. Preview tries to run all the files.

**Comment by cscheid** [comment]
Unfortunately that's not something we can easily fix, because that one-file preview _does_ need to run the other files. Specifically, it needs to get fresh sidebar information like the titles of the other pages.

**Comment by royfrancis** [comment]
The downside of this is that preview becomes unusable in a large collaborative project where multiple authors are working on different files in different states of completeness. I would only care about the one file that I am working on at that point in time while others work on other files.

**Comment by allefeld** [comment]
@cscheid, okay, but couldn't there be an option to simply leave the sidebar information as it is? When I work on one file, getting a quick preview is more important than every detail being correct.

**Comment by cscheid** [comment]
I promise you, folks, we've thought the options through, we know that the situation isn't great right now (we maintain and write quarto.org, which suffers from the same problem!) and we see your votes here. But we have no great way of disentangling the code paths as they exist, and this large amount of work would get in the way of other fixes and improvements that are higher on our priorities.

We appreciate the enthusiasm, but we ask for your trust in our process as well.
